Update PETSc to 3.20.3, update libmesh

cticenhour opened the pull request:
Closes #26461

cticenhour committed:
Update PETSc to 3.20.3

Refs #26461

cticenhour commented:
@roystgnr I'd like to get one round of testing in on this before piling more on, but would you be interested in a libMesh update going in alongside this?

moosebuild commented on petsc:
```diff
@@ -1 +1 @@
-Subproject commit 9a61b8cbbacd10b14c2a18c9bc314b0ecb8ec018
+Subproject commit 1580af9fcba34756b20ac647dbc5659db3fdd05b
```
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1973886/) on d7db356 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1973886/) on d7db356 wanted to post the following:

A change of the following file(s) triggered this check:

  petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The PETSc submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1973886/) on d7db356 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Test timings](https://civet.inl.gov/job/1974131/) on d7db356 wanted to post the following:

View timings [here](https://mooseframework.inl.gov/docs/PRs/26462/timing)

This comment will be updated on new commits.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1973882/) on d7db356 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/26462/site)

This comment will be updated on new commits.

roystgnr commented:
> would you be interested in a libMesh update going in alongside this?

Probably not.  I've got a minor fix that would be useful in MOOSE, but not worth the hassle alone, and I've got a major branch that isn't likely to be ready to merge before the end of the week.

cticenhour commented:
Alright - will ping again if this ends up lingering through Friday or Monday.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1973926/) on d7db356 wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/26462/coverage/xfem)

</details>



This comment will be updated on new commits.

lindsayad commented:
I would be interested in a libmesh update

lindsayad commented:
Can't build https://github.com/idaholab/moose/pull/26477 without a libmesh update. A couple cool new things in libMesh directly leveraged in that MOOSE PR are `L2_HIERARCHIC_VEC` and the ability to form a callback post linear-solve, right before the line search.

That being said, I would be willing to wait for @roystgnr's major branch to merge (is it libmesh/libmesh#3759 or something else?)

roystgnr committed:
Newsletter entries for libMesh submodule update

roystgnr commented:
I've pushed newsletter docs to https://github.com/roystgnr/moose/commit/1bfae7deda9d0f2cc451ba84a81dd5ab1615c393 if you want to cherry-pick that to go alongside a libmesh submodule update.

cticenhour commented:
Thanks for your input! I will update this PR tomorrow to include a libMesh update as well. The only thing holding this back on testing at the moment is a FBLASLAPACK download issue during PETSc alt container creation....working with Logan on it.

cticenhour committed:
Update libmesh submodule

Refs #26461

cticenhour commented on modules/doc/content/newsletter/2024/2024_01.md:
```diff
@@ -9,6 +9,34 @@ for a complete description of all MOOSE changes.
 
 ## libMesh-level Changes
 
+### `2024.01.04` Update
```
```suggestion
### `2024.01.05` Update
```

cticenhour committed:
Tweak date in news update

moosebuild commented on libmesh:
```diff
@@ -1 +1 @@
-Subproject commit 7c03afd53c9c7fd7fcb6e3e20bae37cf1b124986
+Subproject commit 628e9357acd47e403d2452155d68ca717ec5bbe1
```
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1981610/) on 3af4ce9 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1981610/) on 3af4ce9 wanted to post the following:

A change of the following file(s) triggered this check:

  petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The PETSc submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1981610/) on 3af4ce9 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1981610/) on 3af4ce9 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1981610/) on 3af4ce9 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

roystgnr commented:
If there's more futzing to be done anyway, it wouldn't hurt to get libMesh up to https://github.com/libMesh/libmesh/commit/3fcc7e7099f1b4abee4a048fee27a1f01f5e8b18 - the additional changelog entries would be:

- Fixes for VTK .pvtu reading, and preservation of libMesh node, element, and subdomain ids written to VTK files
- `Elem::volume()` no longer can throw an exception when called on an element without an invertible mapping; hence `print_info()` and `get_info()` are now safe to use when emitting debugging messages about twisted geometric elements.

IMHO neither of those are super high priority if you just want to focus on getting the PETSc issues fixed and getting this in.

cticenhour commented:
@roystgnr The GCC min / PETSc-alt failures are all tied to the FBLASLAPACK download issue (which is preventing the build of a new container for those tests), but the new app failures I am seeing are due to the libMesh additions. Do you mind taking a look today? They all seem to be related to the nodeset changes or similar.

roystgnr commented:
Will do!  Looks like we've got a segfault in a Malamute test and 7 (looks like 2 independent) exodiffs in Bison tests?

roystgnr commented:
The good news is that I understand the MALAMUTE failure now.

The bad news is that the MALAMUTE failure is coming from MOOSE doing something unsupported and, up until now, getting away with it via dumb luck.

That input deck creates a mesh of `QUAD8` elements.  It then has MOOSE create Edge3 elements along one side of some of them, because `formulation = mortar`, and MOOSE manually does a `edge3->set_interior_parent(quad8)` on each.  It then has MOOSE convert all the Quad8 elements to Quad9, because `second_order = true`, and the trouble with this of course is that *you can't actually convert one class into another in C++*.  What libMesh does to "convert" a mesh like this is to delete all the lower-order elements and allocate higher-order elements to replace them.  This has apparently (probably?) been working out for MALAMUTE in the past because the allocated elements have been ending up at the same addresses in memory!?  And then those `interior_parent` pointers stay valid.  But nothing guarantees that happening, and any kind of modification of the `all_second_order` code (much less my recent extensive modifications) can stop it from happening. (as could a change of compiler, or of compiler flags, or of day of the week...)

I'm trying to figure out a way to make this robust.

lindsayad commented:
Where is MOOSE doing this? Doesn’t moose rely on libMesh’s api for converting the mesh to second order?

lindsayad commented:
Haven’t you been doing a bunch of work in these mesh order upgrade methods? Is it possible the behavior is changing in this update? It would seem like quite a coincidence for this to have been working for all CI runs for months

roystgnr commented:
> Where is MOOSE doing this? Doesn’t moose rely on libMesh’s api for converting the mesh to second order?

The `set_interior_parent()` calls are in `LowerDBlockFromSidesetGenerator`, I believe.  The `all_second_order()` call is in `SetupMeshAction`.  It is using the libMesh API, but the problem is that even the libMesh API can't guarantee that deallocation and reallocation of objects gives you back the same pointers if only you do it in just the right order.  Maybe there's a way to do it with fancy C++ tricks, if we destruct the existing `Elem` without freeing its memory and then you use placement-`new` to put the new `Elem` in its place (assuming all `Elem` subclasses are exactly the same size, which maybe?), but we don't do that.  We just `delete` the existing `Elem` and `new` a replacement, and the memory allocator has apparently been returning the same pointer, but it's not at all guaranteed to.

> Haven’t you been doing a bunch of work in these mesh order upgrade methods? Is it possible the behavior is changing in this update? It would seem like quite a coincidence for this to have been working for all CI runs for months

It's definitely *triggered* by this update (something is performing another allocation in between the `delete` and the `new`? I'm now doing the `new` before the `delete`?), but it's not *caused* by this update.  I like the "nasal demons" joke about Undefined Behavior, where a compiler is allowed to respond to UB by making demons come out your nose, but really the most insidious allowed response to UB is the one we're seeing here: it's allowed to work the way you hoped right up until for no apparent reason it stops working.

On the bright side, the refactoring should at least make it possible to fix every case of this all in one place.

On the not-so-bright side, the fix is going to make some of my new capabilities asymptotically slower.  We can't safely upgrade O(N_range) elements until after we first check O(N_elem) elements to see if they have any `interior_parent()` links that need to be moved as part of the upgrade.  I don't think anybody's going to need them to be `O(N_range)` but it would have been nice to have had the *option*.

cticenhour committed:
Update libmesh submodule

Refs #26461

cticenhour commented:
@roystgnr FYI - libMesh is now up to https://github.com/libMesh/libmesh/commit/3fcc7e7099f1b4abee4a048fee27a1f01f5e8b18 and I've rebased on current devel.

moosebuild commented on libmesh:
```diff
@@ -1 +1 @@
-Subproject commit 7c03afd53c9c7fd7fcb6e3e20bae37cf1b124986
+Subproject commit 3fcc7e7099f1b4abee4a048fee27a1f01f5e8b18
```
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1988819/) on 2d7f4ce wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1988819/) on 2d7f4ce wanted to post the following:

A change of the following file(s) triggered this check:

  petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The PETSc submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1988819/) on 2d7f4ce wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1988819/) on 2d7f4ce wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1988819/) on 2d7f4ce wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

roystgnr commented:
The Bison failure is independent of the Malamute failure, so this libMesh upgrade had *two* test-breaking bugs.  (insert Simpsons meme here: "two test-breaking bugs *so far*")

It's definitely https://github.com/libMesh/libmesh/pull/3752 triggering the Bison regressions, but I've only figured that out because they're easy to reproduce and bisect, not because I've figured out a cause.

Good news is that the outputs are barely regressions.  Here's the part of the diff being complained about:
```
<  node_ns1 = 1, 2, 5, 6, 9, 11, 13, 15, 16, 19, 20, 23, 24, 27, 29, 31 ;
---
>  node_ns1 = 1, 2, 5, 6, 9, 9, 11, 11, 13, 13, 15, 16, 19, 20, 23, 24, 27, 29, 31 ;
```
The careful observer will note that these nodesets are actually identical as *sets*, but apparently both ExodusII and libMesh internals are happy to write redundancies into a vector representing a set, and even Paraview is happy to retain the redundancies, and fortunately exodiff is paranoid enough to yell at us about them.

Bad news is that, although I'm still trying to track down where the redundancies are coming from, my best guess right now is that we've got dangling pointers to the discarded copies of nodes merged via a mesh stitching, in which case "barely regressions" in the files are covering up "more horrible Undefined Behavior waiting to wreck us" in RAM.

roystgnr commented:
It's not horrible UB this time.  The same node is getting added with the same id twice to the multimap we use.  I really need to change the data structure there; we do have an awful `get_nodeset_map()` raw accessor so it'd technically be an API change, but MOOSE is only using it via a reference-to-auto and hopefully apps are being equally hands-off...

But in the meantime, it looks like I may have just flubbed something in my changes to `BoundaryInfo::clear_stitched_boundary_side_ids()`.  Checking now.

cticenhour committed:
Update libmesh submodule

Refs #26461

cticenhour committed:
Update versioner hashes

Refs #26462

moosebuild commented on libmesh:
```diff
@@ -1 +1 @@
-Subproject commit 7c03afd53c9c7fd7fcb6e3e20bae37cf1b124986
+Subproject commit 8acf1f5507f1ae4a2b6e6e1278265b392321d9dd
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2006974/) on 12db387 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2006974/) on 12db387 wanted to post the following:

A change of the following file(s) triggered this check:

  petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The PETSc submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2006974/) on 12db387 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2006974/) on 12db387 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2006974/) on 12db387 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

roystgnr commented:
Damn, even while sick I should have known to check on this.  Those SCHEMADIFF bugs are from the new field in our VTK output.  I'll see about regolding and verifying and pushing updates.

roystgnr committed:
Update gold VTK files to add new libmesh_node_id

This should make us pass the schemadiffs with the new libMesh submodule;
looking at diffs manually I don't see any other changes.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2014658/) on c028774 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2014658/) on c028774 wanted to post the following:

A change of the following file(s) triggered this check:

  petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The PETSc submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2014658/) on c028774 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2014658/) on c028774 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2014658/) on c028774 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

roystgnr committed:
One more regolded VTK for new libMesh

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2016237/) on 872755a wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2016237/) on 872755a wanted to post the following:

A change of the following file(s) triggered this check:

  petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The PETSc submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2016237/) on 872755a wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2016237/) on 872755a wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2016237/) on 872755a wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour commented:
@roystgnr Just took a peek at the current in-progress testing - framework is now in good shape, but one more test to fix in modules related to IGA (`tensor_mechanics/examples/cframe_iga.cframe`).

roystgnr committed:
Regold C-frame IGA test for new libMesh VTK schema

roystgnr commented:
Well, that was a terrifying red herring.  I reran the example, copied the files to gold/, looked at `git diff` before committing, saw a *huge* diff in the values section, panicked over which of my IGA bugfixes over the past few months could possibly have required a value regolding but not already *prompted* a value regolding, dove into git logs ... then took another look at the `git diff`, which showed a bunch of stuff changing in like the 17th digit, ordinary FP error.  I copied *just* the `libmesh_node_id` array into the old gold, reran the example, and it passed the more-floating-point-friendly VTK XMLDiff just fine.

moosebuild commented on libmesh:
```diff
@@ -1 +1 @@
-Subproject commit 7c03afd53c9c7fd7fcb6e3e20bae37cf1b124986
+Subproject commit 8acf1f5507f1ae4a2b6e6e1278265b392321d9dd
```
**Caution!** This contains a submodule update

roystgnr replied:
Hell yeah it does

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2016681/) on 6d40dfd wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2016681/) on 6d40dfd wanted to post the following:

A change of the following file(s) triggered this check:

  petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The PETSc submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2016681/) on 6d40dfd wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2016681/) on 6d40dfd wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2016681/) on 6d40dfd wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

roystgnr reviewed (approved):
I can't speak to the PETSc part of the change, but everything on the libMesh side should be ready to go.

loganharbour reviewed (changes requested):
Going to add in a minor apptainer change and one more libmesh bump.

loganharbour committed:
Update libmesh submodule

loganharbour committed:
Update versioner hashes

roystgnr commented:
> everything on the libMesh side should be ready to go.

I'd like to thank Nemesis for only striking me down with a new bug report from Alex, and not with the lightning bolt that my hubris above may have deserved.

I'm looking into it now.

moosebuild commented on libmesh:
```diff
@@ -1 +1 @@
-Subproject commit 7c03afd53c9c7fd7fcb6e3e20bae37cf1b124986
+Subproject commit 31948b018e9bea83c138035e952d48065458ba4a
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2019422/) on 7163db6 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2019422/) on 7163db6 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2019422/) on 7163db6 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2019422/) on 7163db6 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

loganharbour commented on scripts/tests/versioner_hashes.yaml:
```diff
@@ -189,3 +189,9 @@ d5ed287fadcea6167a6c9d75924f1c11007ab51d: #26383
   libmesh: bad5b94
   moose-dev: f8a2f21
   wasp: 91987a3
+668ed3ad53d35c108044f8f742507e07a278d48f: #26462
```
Sad, silly mistake

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2021298/) on fbd5908 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2021298/) on fbd5908 wanted to post the following:

A change of the following file(s) triggered this check:

  petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The PETSc submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2021298/) on fbd5908 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2021298/) on fbd5908 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2021298/) on fbd5908 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour commented:
@milljm Could you please give us another set of eyes for these changes? Thanks!

milljm reviewed (approved):
For the Conda stuff, all looks well!

roystgnr commented:
The fix for that last remaining libMesh bug is in https://github.com/libMesh/libmesh/pull/3772

After we get that to master (hopefully tomorrow), we can update the submodule here and we'll have taken care of everything outstanding I know about.

I'm thinking my approval for merging as-is still stands, though, if anyone is chomping at the bit.  The updates in this PR right now include fixes for potentially more serious and definitely more wide-ranging bugs, in long-standing use cases where as far as I can tell we've so far been saved from data corruption by kindly compilers and dumb luck, whereas the bug fixed in https://github.com/libMesh/libmesh/pull/3772 only affects Tet14 refinement, a relatively new feature.

lindsayad commented:
so we didn't get Roy's last fix in?

roystgnr commented:
Nope; sorry.  Hope you're okay working from a manual submodule update until next month.

roystgnr commented:
Looks like we missed the `LIBMESH_REV` update in `docker_ci/Dockerfile` here?  And yet that didn't cause any problems for users?

loganharbour commented:
> Looks like we missed the `LIBMESH_REV` update in `docker_ci/Dockerfile` here? And yet that didn't cause any problems for users?

I need to remove the `docker_ci` folder; it's not used anymore.
